conans/client/generators/cmake_common.py

```diff
@@ -524,7 +530,7 @@ class CMakeCommonMacros:
                        conan_error_compiler_version()
                     endif()
                 else()
-                    if(NOT ${VERSION_MAJOR} VERSION_EQUAL ${CONAN_COMPILER_VERSION})
+                    if(NOT ${VERSION_MAJOR} VERSION_EQUAL ${CONAN_COMPILER_MAJOR})
```

Comment:
I think CONAN_COMPILER_MAJOR doesn't exist here, is this a bug?

Reply:
right, I've moved `conan_split_version` to the upper scope